### PR TITLE
[LBSE] Fix MotionMark repaint issues

### PR DESCRIPTION
**Pull request by nikolaszimmermann**
#### a78e4b31c4407220c6b34773a707e9880c713edd
<pre>
[LBSE] Fix MotionMark repaint issues
<a href="https://bugs.webkit.org/show_bug.cgi?id=270103">https://bugs.webkit.org/show_bug.cgi?id=270103</a>

Reviewed by Rob Buis.

In LBSE transform changes do not result in relayouts, instead repaints
are triggered. That failed to repaint the old position of an object
before the transform change, only the new position - fix that.

Covered by existing tests, when LBSE is active.

* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/js-late-marker-and-object-creation-expected.png:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/js-late-marker-creation-expected.png:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/js-late-pattern-and-object-creation-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/marker-orient-auto-expected.png:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/non-scaling-stroke-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/pattern-deep-referencing-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/pattern-incorrect-tiling-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/pattern-referencing-preserve-aspect-ratio-expected.png: Copied from LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/js-late-marker-and-object-creation-expected.png.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/pattern-rotate-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/pattern-scaled-pattern-space-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/pattern-scaling-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/pattern-skew-transformed-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/pattern-with-transformation-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/shapes-supporting-markers-expected.png:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/transformed-pattern-clamp-svg-root-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/repaint/foreign-object-repainting-after-modifying-container-transform-repaint-rects-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/repaint/foreign-object-repainting-after-modifying-transform-repaint-rects-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/repaint/image-repainting-after-modifying-container-transform-repaint-rects-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/repaint/image-repainting-after-modifying-transform-repaint-rects-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/repaint/shape-repainting-after-modifying-container-transform-repaint-rects-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/repaint/shape-repainting-after-modifying-transform-repaint-rects-expected.txt:
* Source/WebCore/rendering/ReferencedSVGResources.cpp:
(WebCore::CSSSVGResourceElementClient::resourceChanged):
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::repaintRendererOrClientsOfReferencedSVGResources const):
(WebCore::RenderElement::repaintOldAndNewPositionsForSVGRenderer const):
* Source/WebCore/rendering/RenderElement.h:

Canonical link: <a href="https://commits.webkit.org/275702@main">https://commits.webkit.org/275702@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/81a8745176acae0de74caa2622d66e61c6f6b320

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41568 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20582 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43946 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44137 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37660 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/43875 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23691 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17912 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34372 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42142 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17510 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35809 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15032 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15224 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36816 "Passed tests") | [⏳ 🛠 wpe-skia ](https://ews-build.webkit.org/#/builders/WPE-Skia-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37756 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37138 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45508 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16383 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13457 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40889 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18002 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/39299 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/9519 "Build was cancelled. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Checked out pull request; Reviewed by Rob Buis; Compiled WebKit (cancelled)") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18058 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17646 "Built successfully") | | |

### CODE REVIEW

File: Source/WebCore/rendering/RenderElement.cpp [hunk 1]
```diff
@@ -2227,6 +2227,37 @@ void RenderElement::repaintClientsOfReferencedSVGResources() const
     if (auto* enclosingResourceContainer = lineageOfType<RenderSVGResourceContainer>(*this).first())
         enclosingResourceContainer->repaintAllClients();
 }
+
+void RenderElement::repaintOldAndNewPositionsForSVGRenderer() const
```
Comment (rwlbuis) [inline Source/WebCore/rendering/RenderElement.cpp:2231]: I don't like the name much but I can't think of anything better.

**Review by rwlbuis — COMMENTED** [review]
The patch looks good to me but maybe Simon has some comments.

**Comment by nikolaszimmermann** [comment]
Will land it today, if Simon sees any issues, we can adress them in a follow-up.

**Review by rwlbuis — APPROVED** [review]
LGTM.

**Comment by nikolaszimmermann** [comment]
Oops, sorry @Ahmad-S792 didn't see that you already fixed the mq.

**Comment by Ahmad-S792** [comment]
> Oops, sorry @Ahmad-S792 didn't see that you already fixed the mq.

Happy to help. 👍

**Pull request merged**